File: queue/src/main/java/br/com/queue/dtos/user/metrics/ResponseUserDashBoardDto.java
package br.com.queue.dtos.user.metrics;

import java.util.List;

public record ResponseUserDashBoardDto(

        ResponseCountTotalUsersStatisticsDto countTotalUsersStatistics,

        ResponseUserPercentagesStatisticsDto userPercentagesStatistics,

        List<ResponseUsersCreatedByMonthStatisticsDto> usersCreatedByMonthStatistics,

        List<ResponseServicesByUserStatisticsDto> countServicesByUsers,

        List<ResponseUsersByRoleStatisticsDto> countRoleByUsers



) {
}

File: queue-frontend/src/app/dtos/users/statistics/ResponseUserDashBoardDto.ts
import { ResponseCountTotalUsersStatisticsDto } from "./ResponseCountTotalUsersStatisticsDto";
import { ResponseServicesByUserStatisticsDto } from "./ResponseServicesByUserStatisticsDto";
import { ResponseUserPercentagesStatisticsDto } from "./ResponseUserPercentagesStatisticsDto";
import { ResponseUsersByRoleStatisticsDto } from "./ResponseUsersByRoleStatisticsDto";
import { ResponseUsersCreatedByMonthStatisticsDto } from "./ResponseUsersCreatedByMonthStatisticsDto";

export interface ResponseUserDashBoardDto{

  countTotalUsersStatistics: ResponseCountTotalUsersStatisticsDto;
  userPercentagesStatistics: ResponseUserPercentagesStatisticsDto;
  usersCreatedByMonthStatistics: ResponseUsersCreatedByMonthStatisticsDto[];
  countServicesByUsers: ResponseServicesByUserStatisticsDto[];
  countRoleByUsers: ResponseUsersByRoleStatisticsDto[];
}
